### ISSUE TITLE
AzureMLPipelineDataSet does not play well with other datasets

### ISSUE THREAD

**Issue by marrrcin** [getindata/kedro-azureml#56]
@tomasvanpottelbergh this is related to the `AzureMLPipelineDataSet` you've implemented.

---
## Issue
AzureMLPipelineDataSet does not work with i.e. `AzureMLPandasDataSet` when pipeline data passing is enabled.
## Steps to reproduce:
1. Use spaceflights with the following catalog:
```yaml
companies:
  type: pandas.CSVDataSet
  filepath: data/01_raw/companies.csv
  layer: raw

reviews:
  type: pandas.CSVDataSet
  filepath: data/01_raw/reviews.csv
  layer: raw

shuttles:
  type: pandas.ExcelDataSet
  filepath: data/01_raw/shuttles.xlsx
  layer: raw

# TODO: this does not work
preprocessed_companies:
  type: kedro_azureml.datasets.pandas_dataset.AzureMLPandasDataSet
  azureml_dataset: e2e_tests_preprocessed_companies

preprocessed_shuttles:
  type: kedro_azureml.datasets.AzureMLPipelineDataSet
  dataset:
    type: pandas.CSVDataSet
    filepath: "shuttles.csv"
```
2. Enable `pipeline_data_passing` feature in `azureml.yml`
```yaml
azure:
    # ...
    pipeline_data_passing:
        enabled: true
```
3. Run the pipeline (I'm using docker flow, but it should not matter).

### Expected result
Pipeline executes normally.

### Actual result
Job fails as soon as the `create_model_input_table_node` starts. 

```
Error Code: ScriptExecution.StreamAccess.NotFound
Native Error: error in streaming from input data sources
	StreamError(NotFound)
=> stream not found
	NotFound
Error Message: The requested stream was not found. Please make sure the request uri is correct.| session_id=fb6a7934-81ac-4947-b2d8-d39bdcc4fad8
```

I suspect that the **pipeline_data_passing** feature assumes that there will **always** be something in the `--az-input` locations, while that is not always the case in Kedro, as users **CAN** specify intermediate data anywhere in the catalog.

![image](https://user-images.githubusercontent.com/6958772/235163706-2edcc220-90cc-463b-b420-53eaa9958bb7.png)

**Comment by marrrcin** [comment]
Looking forward to see that!

**Comment by marrrcin** [comment]
Closed by #60. V1 datasets are marked as deprecated.
